android/src/main/java/com/mattermost/networkclient/helpers/UploadRequestBody.kt

```diff
@@ -0,0 +1,50 @@
+package com.mattermost.networkclient.helpers
+
+import android.net.Uri
+import com.facebook.react.bridge.ReactContext
+
+import okhttp3.MediaType
+import okhttp3.MediaType.Companion.toMediaTypeOrNull
+import okhttp3.RequestBody
+import okhttp3.internal.closeQuietly
+import okio.*
+import java.io.IOException
+
+
+class UploadFileRequestBody(private val reactContext: ReactContext, private val uri: Uri, private val skipBytes: Long, private val listener: ProgressListener) : RequestBody() {
+
+    private val stream = reactContext.contentResolver.openInputStream(uri)!!
+    private val total: Double = stream.available().toDouble();
+
+    override fun contentLength(): Long {
+        return total.toLong();
```

Comment:
Is this value always available?

Reply:
Yup; I grab the stream's length and put it in the `total` val; though that's assuming there is a stream - otherwise it returns `content-length: 0`

Reply:
So make sure you don't divide by 0